src/RamOperation.h

```diff
@@ -696,11 +693,11 @@ class RamReturn : public RamOperation {
 
     /** Create clone */
     RamReturn* clone() const override {
-        auto* res = new RamReturn();
+        std::vector<std::unique_ptr<RamValue>> newValues;
         for (auto& cur : values) {
-            res->values.emplace_back(cur->clone());
+            newValues.push_back(std::unique_ptr<RamValue>(cur->clone()));
```

Comment:
Since you're putting a RamValue into a vector of unique pointers to RamValues, you can just do
```suggestion
            newValues.emplace_back(cur->clone());
```